### PR TITLE
Upgrade HTTPListRedirects to py3

### PR DESCRIPTION
**Pull request by anas-yousef**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Description
Made the script python3 compatible and fixed a small bug where we did not save the response of one of the requests

## Screenshots
image
image

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

### CODE REVIEW

**Review by tkatzir — APPROVED** [review]
Great!

File: Packs/CommonScripts/Scripts/HTTPListRedirects/HTTPListRedirects.py [hunk 1]
```diff
@@ -5,27 +5,54 @@
...
+    urls = []
+    if response.history:
+        for resp in response.history:
+            urls.append({'Data': resp.url, 'Status': resp.status_code})
+    urls.append({'Data': response.url, 'Status': response.status_code})
+    return urls
+
+
+def delete_environment_variables(use_system_proxy: str):
+    if use_system_proxy == 'false':
```
Comment (tkatzir) [inline Packs/CommonScripts/Scripts/HTTPListRedirects/HTTPListRedirects.py:34]:
```suggestion
    if use_system_proxy.lower() == 'false':
```

File: Packs/CommonScripts/Scripts/HTTPListRedirects/HTTPListRedirects.py [hunk 2]
```diff
@@ -5,27 +5,54 @@
...
+        for env_var in env_variables:
+            if(env_var in os.environ):
+                del os.environ[env_var]
+
+
+def main():
+    url = demisto.args().get('url')
+    try:
+        delete_environment_variables(use_system_proxy=demisto.args().get('use_system_proxy'))
+        url = f'http://{url}' if not url.lower().startswith('http') else url
```
Comment (tkatzir) [inline Packs/CommonScripts/Scripts/HTTPListRedirects/HTTPListRedirects.py:45]: Question: If `trust_any_certificate` is `True`, doesn't it imply that it's https?

File: Packs/CommonScripts/Scripts/HTTPListRedirects/HTTPListRedirects.py [hunk 3]
```diff
@@ -5,27 +5,54 @@
...
+
+        response = get_response(url=url,
+                                use_head=demisto.args()['useHead'],
+                                verify_ssl=demisto.args().get('trust_any_certificate') != 'true')
```
Comment (tkatzir) [inline Packs/CommonScripts/Scripts/HTTPListRedirects/HTTPListRedirects.py:49]:
```suggestion
                                verify_ssl=demisto.args().get('trust_any_certificate', 'true').lower() != 'true')
```
Assuming `demisto.args().get(...)` supports default values.
In any case, it's better to `lower()` the value, so we're sure we're in the same case.

File: Packs/CommonScripts/Scripts/HTTPListRedirects/HTTPListRedirects.py [hunk 4]
```diff
@@ -5,27 +5,54 @@
...
+        history_urls = get_response_history(response=response)
+        demisto.results(create_command_result(history_urls=history_urls))
+
+    except Exception as e:
+        return_error(f'Failed to execute script.\nError:\n{str(e)}')
```
Comment (tkatzir) [inline Packs/CommonScripts/Scripts/HTTPListRedirects/HTTPListRedirects.py:54]:
```suggestion
        return_error(f'Failed to execute script. Error:\n{str(e)}')
```
Too many newlines...

**Review by tkatzir — APPROVED** [review]
Looks great!
Some minor comments and questions.
Approved.

File: Packs/CommonScripts/Scripts/HTTPListRedirects/HTTPListRedirects.py (hunk 2 above)
Comment (anas-yousef) [reply Packs/CommonScripts/Scripts/HTTPListRedirects/HTTPListRedirects.py:45]: The opposite, if *trust_any_certificate* is *True*, that means we don't want to use the *SSL Protocol* to encrypt the request which means it's not *https* (Correct me if I'm wrong)

Comment (tkatzir) [reply Packs/CommonScripts/Scripts/HTTPListRedirects/HTTPListRedirects.py:45]: I don't think it's related.
`trust_any_certificate` is relevant only with HTTPS - It tells the client to verify the server's certificate.
It's irrelevant with HTTP.

**Pull request merged**